flake.nix

```diff
@@ -44,11 +45,27 @@
           devShells.ci-linter = util.wrapShell pkgs.mkShellNoCC { packages = util.linters; };
 
           devShells.ci_clang18 = util.wrapShell (util.mkShellWithCC pkgs.clang_18) { packages = [ pkgs.python3 ]; };
+          devShells.ci_clang19 = util.wrapShell (util.mkShellWithCC inputs.nixpkgs2411.legacyPackages.${system}.clang_19) { packages = [ pkgs.python3 ]; };
           devShells.ci_gcc48 = util.wrapShell (util.mkShellWithCC pkgs.gcc48) { packages = [ pkgs.python3 ]; };
           devShells.ci_gcc49 = util.wrapShell (util.mkShellWithCC pkgs.gcc49) { packages = [ pkgs.python3 ]; };
           devShells.ci_gcc7 = util.wrapShell (util.mkShellWithCC pkgs.gcc7) { packages = [ pkgs.python3 ]; };
           devShells.ci_gcc11 = util.wrapShell (util.mkShellWithCC pkgs.gcc11) { packages = [ pkgs.python3 ]; };
           devShells.ci_gcc14 = util.wrapShell (util.mkShellWithCC pkgs.gcc14) { packages = [ pkgs.python3 ]; };
+
+          devShells.ci_valgrind_clang14 = util.wrapShell (util.mkShellWithCC pkgs.clang_14) { packages = [ pkgs.valgrind ]; hardeningDisable = [ "fortify" ]; };
```

Comment:
I'm not sure this needs a separate shell; it's a bug in nix that causes FORTIFY_SOURCE to be set unconditionally, right? Normally it wouldn't be, so I'd suggest to unset it in the default shells and add a pointer to the respective nix issue.

Reply:
The main reason for the extra shell is the additional valgrind dependency. The fortify stuff indeed needs to be added to the other shells as well - I will do that shortly in a separate PR.

We could install valgrind always if you insist - then we need half the shells.

Reply:
Yes, always having valgrind makes sense to me

Reply:
Okay, I've added it everywhere and removed the extra shells.